daemon/daemon_unix.go

```diff
@@ -1757,7 +1757,23 @@ func (daemon *Daemon) setupSeccompProfile() error {
 }
 
 func (daemon *Daemon) useShimV2() bool {
-	return cgroups.IsCgroup2UnifiedMode()
+	env := os.Getenv("MOBY_DISABLE_SHIM_V2")
+	if env == "" {
+		return true
+	}
+	// MOBY_DISABLE_SHIM_V2 is deprecated from its birth.
+	// MOBY_DISABLE_SHIM_V2 is just added as a workaround for potential regression issues.
+	logrus.Warn("MOBY_DISABLE_SHIM_V2 is deprecated and planned to be removed in future releases")
+	disableV2, err := strconv.ParseBool(env)
+	if err != nil {
+		logrus.Warnf("Ignoring invalid MOBY_DISABLE_SHIM_V2 value %q: %v", env, err)
```

Comment:
Inclined to make this a hard failure as well. `strconv.ParseBool` already accepts a large range of values that "work", so might be better to fail and have the user correct the invalid value

Reply:
updated